benchmarks/nel/scripts/datasets/dataset.py

```diff
@@ -0,0 +1,300 @@
+            "nlp_best": root_path / "training" / dataset_name / "model-best",
+            "kb": root_path / "temp" / dataset_name / "kb",
+            "corpora": root_path / "corpora" / dataset_name,
+            "entities": assets_path / "entities.pkl",
+            "failed_entity_lookups": assets_path / "entities_failed_lookups.pkl",
+            "annotations": assets_path / "annotations.pkl"
+        }
+
+    @property
+    def name(self) -> str:
+        """ Returns dataset name.
+        """
+        raise NotImplementedError
+
+    def create_knowledge_base(self, model_name: str, **kwargs) -> None:
+        """ Creates and serializes knowledge base.
+        vectors_model (str): Name of model with word vectors to use.
+        """
+
+        self._nlp_base = spacy.load(model_name, exclude=["tagger", "lemmatizer", "attribute_ruler"])
+        print("Parsing external corpus")
+        self._entities, self._failed_entity_lookups, self._annotations = self._parse_external_corpus(**kwargs)
+
+        print(f"Constructing knowledge base with {len(self._entities)} entries")
+        self._kb = KnowledgeBase(vocab=self._nlp_base.vocab, entity_vector_length=self._nlp_base.vocab.vectors_length)
+        entity_list: List[str] = []
+        freq_list: List[int] = []
+        vector_list: List[numpy.ndarray] = []
+        for qid, info in self._entities.items():
+            entity_list.append(qid)
+            freq_list.append(info["frequency"])
+            vector_list.append(self._nlp_base(info["description"]).vector)
+        self._kb.set_entities(entity_list=entity_list, vector_list=vector_list, freq_list=freq_list)
+        for qid, info in self._entities.items():
+            for name in info["names"]:
+                self._kb.add_alias(alias=name.replace("_", " "), entities=[qid], probabilities=[1])
+
+        # Serialize knowledge base & entity information.
+        for to_serialize in (
+            (self._paths["entities"], self._entities),
+            (self._paths["failed_entity_lookups"], self._failed_entity_lookups),
+            (self._paths["annotations"], self._annotations)
+        ):
+            with open(to_serialize[0], 'wb') as fp:
+                pickle.dump(to_serialize[1], fp)
+        self._kb.to_disk(self._paths["kb"])
+        if not os.path.exists(self._paths["nlp_base"]):
+            os.mkdir(self._paths["nlp_base"])
+        self._nlp_base.to_disk(self._paths["nlp_base"])
+        print("Successfully constructed knowledge base.")
+
+    def compile_corpora(self) -> None:
+        """ Creates train/dev/test corpora for Reddit entity linking dataset.
+        """
+
+        self._load_resource("entities")
+        self._load_resource("failed_entity_lookups")
+        self._load_resource("annotations")
+        self._load_resource("nlp_base")
+
+        Doc.set_extension("overlapping_annotations", default=None)
+        self._annotated_docs = self._create_annotated_docs()
+        self._serialize_corpora()
```

Comment:
Should this have a succesful logging message after serializing? Maybe adding the location of where the corpora were stored?